Final post for 596 (#33)
compusciencing/compusciencing.github.io: _posts/2019-12-16-NormandyBryson.md

---
layout: post
title: Transformable Wheel Robot Game To Teach Kids About Research
tags: [Javascript, games, research, matter.js]
author: Normandy Rivver

> include_mathjax: false

Author: Delete what line?  So you don't want the title:, tags:, etc?

> include_mathjax: false

Reviewer: ```suggestion
include_mathjax: false
```

Just that line.

---

## THE BEGINNING

> CSC 596 is a life changing course and I know what you must be thinking “how can a course impact you this much..”Well I am here to share this journey and hopefully show how one course can provide a big boost to not only me but probably any student studying computer science.  One of the issues, or maybe fears, I’ve had with computer science is not knowing what to build, what to try and code, where to start, how to break the problem down, how to find answers to my problems which do not rely on teachers or teaching assistants.  This is a long list of issues and yet CSC 596 not only makes you confront them but helps you overcome them.

Author: Why are some of these words in blue and some in red in both the red version and green version?

Svgs would prove to be the next issue mainly because I decided on using them for terrain generation.  Basically I could draw an svg image and import it as a terrain into the game and while this caused an increase in load times I believed it would simplify the process of creating terrains, reduce the bouncing effect, and allow for easier to make terrains which were different from each other.  While I was able to get one terrain loaded and working successfully getting more terrains which were usable proved not successful.  This lack of success was and is in spite of drawing an svg image directly under the original workable svg image and loading it into the program, changing a number of settings to try and see if this helped, and flipping the original terrain in different directions.  

> ## LOOKING AHEAD

Author: I can do this.  If I do I get an error throughout entire md file about multiple headings.

> ## LOOKING AHEAD

Reviewer: Maybe in your editor, but not on here.

This isn’t to say the project did produce a working prototype but the issues still present and the lack of usable terrain means I would probably go back and just make a randomly generated coded terrain with or just make a few terrains with puzzles to solve by hand.  The final project can be played at the link provided by scrolling down to the bottom of the page and clicking the link:  [Github] (https://github.com/normandyr/csc596) .  Really this class and this one project is just one small example of how 596 as helped me to be a better programmer.  CSC 596 has mainly helped to give me confidence to solve my own programming problems, develop a mindset which approaches problems in a more constructive manner, show how to approach problems to begin with, and to push me toward building my own projects.  

> Currently, I am building a data mining project using Selenium, python request, and python beautifulsoup where I have a product lets say the Google pixel 3 and the Google pixel 4.  My project will open a web browser go to Google enter “X Google Pixel is X”  where the first X is a list of websites which do tech reviews currently have “The Verge” and “Android Central” working and the second X is a list of valence words such as that correspond to positive [“good”, “awesome”, “etc.”] and negative [“bad, “terrible”, “etc.”].  The code will loop through the phone products, loop through the websites listed, and loop through the valence words pulling the comments section from each search and then I’ll do some data mining opinion analysis or bag of words analysis on the information written to the files.  Here is a stack over flow comment I made regarding an issue I had with this problem.  It shows how I approached the issue and how I eventually decided to deal with the issue:

Reviewer: ```suggestion

I am building a data mining project in Python using [Selenium](https://selenium-python.readthedocs.io/), [Requests](https://realpython.com/python-requests/), and [Beautifulsoup](https://www.crummy.com/software/BeautifulSoup/bs4/doc/).  My project will open a web browser go to Google enter “X Google Pixel is X”  where the first X is a list of websites which do tech reviews currently have “The Verge” and “Android Central” working and the second X is a list of valence words such as that correspond to positive [“good”, “awesome”, “etc.”] and negative [“bad, “terrible”, “etc.”].  The code will loop through the phone products, loop through the websites listed, and loop through the valence words pulling the comments section from each search and then I’ll do some data mining opinion analysis or bag of words analysis on the information written to the files.  [Here is a stack overflow comment I made regarding an issue I had with this problem](https://stackoverflow.com/questions/59333437/cant-grab-comment-text-using-selenium-python-requests-and-beautifulsoup-from).  It shows how I approached the issue and how I eventually decided to deal with the issue.
```

> [StackoverFlow] (https://stackoverflow.com/questions/59333437/cant-grab-comment-text-using-selenium-python-requests-and-beautifulsoup-from)

Reviewer: ```suggestion
```

> All skills I learned from Dr. Clark, this class, and the Missouri State University Computer Science program.

Reviewer: ```suggestion
```
